src/ctr_logging.c

```diff
@@ -325,6 +345,12 @@ static int write_journald(int pipe, char *buf, ssize_t buflen)
 		/* per docker journald logging format, CONTAINER_PARTIAL_MESSAGE is set to true if it's partial, but otherwise not set. */
 		if (partial && writev_buffer_append_segment_no_flush(&bufv, "CONTAINER_PARTIAL_MESSAGE=true", PARTIAL_MESSAGE_EQ_LEN) < 0)
 			return -1;
+		if (container_labels) {
+			for (int i = 0; i < container_labels_count; i++) {
+				if (writev_buffer_append_segment_no_flush(&bufv, container_labels[i], container_labels_lengths[i]) < 0)
+					return -1;
+			}
```

Comment:
I suggest to not create the arrays before and then have to worry about cleaning them up.

Just strdup container_labels here and call `writev_buffer_append_segment_no_flush` while you parse it.

Reply:
Going back to this. Since the array will live for the entire of the program. Is this needed? Not to challenge, just want to get more familiar with C and this codebase.